src/main/java/xyz/kyngs/herbot/handlers/command/AbstractCommand.java

```diff
@@ -1,17 +1,114 @@
 package xyz.kyngs.herbot.handlers.command;
 
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
 import xyz.kyngs.herbot.HerBot;
+import xyz.kyngs.herbot.handlers.command.argument.AbstractArgument;
+import xyz.kyngs.herbot.handlers.command.argument.Arguments;
+import xyz.kyngs.herbot.handlers.user.UserProfile;
+import xyz.kyngs.herbot.util.ExceptionUtil;
+import xyz.kyngs.herbot.util.ImmutableEntry;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
 
 public abstract class AbstractCommand implements CommandExecutor {
 
     protected final HerBot herBot;
     protected final String description;
+    private final List<AbstractArgument> args;
+
+    public String getPermission() {
+        return permission;
+    }
+
+    private final String permission;
 
-    public AbstractCommand(HerBot herBot, String description) {
+    public AbstractCommand(HerBot herBot, String description, String permission) {
         this.herBot = herBot;
         this.description = description;
+        this.permission = permission;
+        args = new ArrayList<>();
     }
 
+    public void addArg(AbstractArgument arg) {
+        args.add(arg);
+    }
+
+    @Override
+    public void onCommand(User author, Guild guild, TextChannel channel, Message message, String[] args, UserProfile profile, GuildMessageReceivedEvent event) {
+        if (!checkPerm(message, profile) || !checkArgSize(message, args) || !checkArgumentType(message, args)) return;
+
+        var list = new ArrayList<ImmutableEntry<String, AbstractArgument>>();
+
+        for (int i = 0; i < this.args.size(); i++) {
+            list.add(new ImmutableEntry<>(args[i], this.args.get(i)));
+        }
+
+        exec(author, guild, channel, message, new Arguments(list, herBot.getJda()), profile, event);
+
+    }
+
+    private boolean checkPerm(Message message, UserProfile profile) {
+        if (!profile.hasPermission(permission)) {
+            var builder = new EmbedBuilder();
+            builder.setColor(Color.RED);
+            builder.setTitle("Tuto akci nemůžeš provést");
+            builder.setDescription("Na tuto akci nemáš dostatečné oprávnění");
+            message.reply(builder.build()).mentionRepliedUser(false).queue();
+            return false;
+        }
+        return true;
```

Comment:
You could change the condition and return true, if the user has permissions. Which would lead into removing the negation (It's always easier to read something without negations).

Reply:
That is exactly how it works rn, if user has permission it returns true, otherwise false.